test/metabase/lib/drill_thru/zoom_in_timeseries_test.cljc

```diff
@@ -134,6 +136,81 @@
       :column-name "sum"
       :expected    {:type :drill-thru/zoom-in.timeseries}})))
 
+(def ^:private metadata-provider-with-orders-created-at-as-date
+  (merged-mock/merged-mock-metadata-provider
+   meta/metadata-provider
+   {:fields [{:id             (meta/id :orders :created-at)
+              :base-type      :type/Date
+              :effective-type :type/Date
+              :semantic-type  :type/CreationDate
+              :database-type  "DATE"}]}))
+
+(defn- zoom-in-timeseries-drill-for-orders-created-at
+  ([column-name temporal-unit]
+   (zoom-in-timeseries-drill-for-orders-created-at meta/metadata-provider column-name temporal-unit))
+  ([metadata-provider column-name temporal-unit]
+   {:drill-type   :drill-thru/zoom-in.timeseries
+    :click-type   :cell
+    :query-type   :aggregated
+    :column-name  column-name
+    :custom-query (-> (lib/query metadata-provider
+                                 (lib.metadata/table metadata-provider (meta/id :orders)))
+                      (lib/aggregate (lib/count))
+                      (lib/aggregate (lib/sum (lib.metadata/field metadata-provider (meta/id :orders :tax))))
+                      (lib/aggregate (lib/max (lib.metadata/field metadata-provider (meta/id :orders :discount))))
+                      (lib/breakout (-> (lib.metadata/field metadata-provider (meta/id :orders :created-at))
+                                        (lib/with-temporal-bucket temporal-unit))))
+    :custom-row   {"count" 77, "sum" 1, "max" nil, "CREATED_AT" "2022-12-01"}
+    :expected     {:type :drill-thru/zoom-in.timeseries}}))
+
+(deftest ^:parallel returns-zoom-in-timeseries-test-4
```

Comment:
A lot of our tests end in `test-n` or whatever as a byproduct of breaking out giant tests into smaller ones, it's even better to give different tests more meaningful names if you can. Something like `returns-zoom-in-timeseries-count-date-by-month-test` or something and then `-sum-date-by-month-test` for the next one is a little bit better than just giving every test a number .

I would really consider combining these all into one test tho, maybe something like

```clj
(deftest ^:parallel returns-zoom-in-timeseries-date-column-test
  (testing "zoom-in.timeseries should be returned for Date dimension (#33811, #39366)"
    (doseq [aggregation ["count" "sum" "max"]
            bucketing   [:month :day]]
      (testing (lib.util/format "aggregation = %s, bucketing = %s" aggregation bucketing)
        (lib.drill-thru.tu/test-returns-drill
         (zoom-in-timeseries-drill-for-orders-created-at
          metadata-provider-with-orders-created-at-as-date
          aggregation
          bucketing))))))
```

otherwise it gets really repetitive and noisy to write out 6 versions of the same test that only differ by one or two things

Reply:
Fixed in latest commit